src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/TaskListTabPanel.tsx

```diff
@@ -522,11 +527,15 @@ const TaskListTabPanel = (props: TaskListTabPanelProps) => {
           <DataGrid
             columns={columns}
             rows={rows}
-            pageSize={5}
             isCellEditable={isCellEditable}
             experimentalFeatures={{ newEditingApi: true }}
             processRowUpdate={processRowUpdate}
-            rowsPerPageOptions={[5]}
+            pageSize={pageSize}
+            rowsPerPageOptions={[5, 10, 15, 100]}
+            onPageSizeChange={(newPageSize) => {
+              localStorage.setItem('tl-table-row-count', String(newPageSize));
```

Comment:
To be consistent with using TABLE_ROW_COUNT variable defined on line 112, can we use it here instead of using 'tl-table-row-count' ? This will also help the reader to understand the code easily!